mite/__main__.py

```diff
@@ -282,6 +283,10 @@ def setup_logging(opts):
         level=opts["--log-level"],
         format="[%(asctime)s] <%(levelname)s> [%(name)s] [%(pathname)s:%(lineno)d %(funcName)s] %(message)s",
     )
+    # add the journey_logging attribute to the object
+    # default is False´
+    journey_logging = opts.get("--journey-logging", False)
+    setattr(logging, "journey_logging", journey_logging)
```

Comment:
this approach seems the simplest and least intrusive and causing the least amount of code changes. It also allows the flag to be used by multiple objects. Passing to GenericStatsOutput may well result in largish code changes

Reply:
There's a bit of a tradeoff between expediency now and maintainability later.  I will leave it to others to also contribute to the discussion, but within the context of python programming, adding our own new attributes to stdlib module objects is sufficiently outside the norm that I don't really think it should be done.

We already have the `opts` object, which gets passed around quite a few places.  This functions as a "global" store of state related to the configuration of the CLI behavior.  I don't anticipate large or intrusive code changes stemming from using that approach for this flag as well.  Concretely, we'd have to pass either `opts` or a boolean value we extract out of it as an additional argument to `_get_http_stats_output` [here](https://github.com/sky-uk/mite/blob/39fbd205838133b7af29d00bbc07251e9038c674/mite/cli/test.py#L138).  That then needs to be passed to the constructor when we instantiate the HttpStatsOutput class in that function, and also the constructor [here](https://github.com/sky-uk/mite/blob/11d047492403227cd722d2466cb9b1ad5ab3d3f0/mite/logoutput.py#L37) should take it in as a new argument.  It requires perhaps half a dozen extra lines of code, but that price is well worth it in my mind to avoid mutating the stdlib `logging` module object.  As I said though, I think we can look to others for a consensus here about how to proceed.

Reply:
upon reflection, i am tending to agree with you, i have looked at the way the logging is instantiated and, as you point out, the addition code change to add opts to the log classes is not a great deal and avoids dynamic additions to objects via the setattr pattern. I will have a play with the code and resubmit :)

Reply:
Thanks, the new code LGTM :slightly_smiling_face: